### PR TITLE
Ember-Core - HeaderToRequestNoBody - Tweak

### PR DESCRIPTION
**Pull request by diesalbla**
- Use exists / forall instead of filter/find/map
- Avoid the call to `String.toUpperCase`, since that would allocate an entire new string that is immediately discarded.

### CODE REVIEW

File: ember-core/shared/src/main/scala/org/http4s/ember/core/h2/PseudoHeaders.scala [hunk 1]
```diff
@@ -70,34 +76,33 @@ private[h2] object PseudoHeaders {
...
     val authority = extractAuthority(pseudos)
 
-    val noOtherPseudo = pseudos
-      .filterNot(t => requestPsedo.contains(t._1))
-      .isEmpty
+    val noOtherPseudo = pseudos.forall(t => requestPsedo.contains(t._1))
 
-    val teIsCorrect = headers.find(_._1 === "te").headOption.fold(true)(_._2 === "trailers")
-    val connectionIsAbsent = headers.find(_._1 === "connection").isEmpty
+    val teIsCorrectAndNoConnection = !headers.exists { case (k, v) =>
```
Comment (danicheg) [inline ember-core/shared/src/main/scala/org/http4s/ember/core/h2/PseudoHeaders.scala:81]: `!exists(_ == foo)` is equal to `forall(_ != foo)`, isn't?

Comment (diesalbla) [reply ember-core/shared/src/main/scala/org/http4s/ember/core/h2/PseudoHeaders.scala:81]: Yeah... it's one of this cases in which one may want to think this in English... Which one would be easier to say?

> All headers must not be neither 1. a connection, nor 2. a "te" without "trailers" 

> No header can be either 1. a connection, or 2. a "te" without "trailers"

The former bends easier to `.forall( case (k, v) => ! (---) && ! (---) ) `, whereas the latter look closer to the code as is.

File: ember-core/shared/src/main/scala/org/http4s/ember/core/h2/PseudoHeaders.scala [hunk 2]
```diff
@@ -59,6 +59,12 @@ private[h2] object PseudoHeaders {
   }
 
   def headersToRequestNoBody(hI: NonEmptyList[(String, String)]): Option[Request[fs2.Pure]] = {
+    def toRawHeader(k: String, v: String): Header.ToRaw =
+      Header.ToRaw.rawToRaw(Header.Raw(org.typelevel.ci.CIString(k), v))
+
+    def isLowerCase(str: String): Boolean =
```
Comment (ChristopherDavenport) [inline ember-core/shared/src/main/scala/org/http4s/ember/core/h2/PseudoHeaders.scala:65]: This one I think may create more allocated objects than it saves us.

Comment (diesalbla) [reply ember-core/shared/src/main/scala/org/http4s/ember/core/h2/PseudoHeaders.scala:65]: Are you referring to the `String.forall` boxing the Char characters?

This is the implementation in the library: 

```Scala
  def forall(@deprecatedName("f", "2.13.3") p: Char => Boolean): Boolean = {
    var i = 0
    val len = s.length
    while(i < len) {
      if(!p(s.charAt(i))) return false
      i += 1
    }
    true
  }
```

Comment (diesalbla) [reply ember-core/shared/src/main/scala/org/http4s/ember/core/h2/PseudoHeaders.scala:65]: The `StringOps` is just a syntax-extension "decorator" class, and it does not inherit from any collection-generic trait that may require a `forall` with an `AnyRef` on it... so it should be unboxed... but only the decompiler can tell, to be sure.

**Pull request closed without merging**